src/Type/Generic/TemplateType.php

```diff
@@ -2,11 +2,20 @@
 
 namespace PHPStan\Type\Generic;
 
+use PHPStan\Type\CompoundType;
 use PHPStan\Type\Type;
 
-interface TemplateType extends Type
+interface TemplateType extends CompoundType
```

Comment:
I don't think this should be done. `ObjectType` isn't compound so TemplateObjectType shouldn't be either.

Reply:
I needed this so that I can make `ObjectType(DateTime)->isSuperTypeOf(TemplateObjectType(of DateTime))` return `No`, while the inverse returns `Yes`.

Reply:
In my understanding, when a `isSuperTypeOf()` receives a `CoumpoundType`, it delegates the computation to `CompoundType::isSubTypeOf()`. This allows a compound type to "take over" the default order, without having all types know about it. Am I right ? This is what I did here.